x-pack/legacy/plugins/maps/public/layers/sources/es_search_source/es_search_source.js

```diff
@@ -232,22 +252,30 @@ export class ESSearchSource extends AbstractESSource {
   // searchFilters.fieldNames contains geo field and any fields needed for styling features
   // Performs Elasticsearch search request being careful to pull back only required fields to minimize response size
   async _getSearchHits(layerName, searchFilters, registerCancelCallback) {
+    const initialSearchContext = {
+      docvalue_fields: []
+    };
     const geoField = await this._getGeoField();
+    const timeFields = await this.getTimeFields();
+    timeFields.forEach(field => {
+      initialSearchContext.docvalue_fields.push({
+        field: field.name,
+        format: 'epoch_millis'
+      });
+    });
 
     let searchSource;
     if (geoField.type === ES_GEO_FIELD_TYPE.GEO_POINT) {
       // Request geo_point and style fields in docvalue_fields insted of _source
       // 1) Returns geo_point in a consistent format regardless of how geo_point is stored in source
       // 2) Setting _source to false so we avoid pulling back unneeded fields.
-      const initialSearchContext = {
-        docvalue_fields: searchFilters.fieldNames
-      };
+      initialSearchContext.docvalue_fields.push(...searchFilters.fieldNames);
```

Comment:
The time field is getting included in docvalue_fields twice. Below is an example request. To prevent this, filter searchFilters.fieldNames to exclude time fields since they are already in docvalue_fields.

```
"docvalue_fields": [
    {
      "field": "@timestamp",
      "format": "epoch_millis"
    },
    "@timestamp",
    "geo.coordinates"
  ],
```